File: DSA/ShatakshiMishra_590015008/Day4Question1.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
    int n;
    cin >> n;
    if(n==0)
    {
        cout << "Array is empty" << endl;
        return 0;
    }

    vector<int> nums(n);
    for (int i = 0; i < n; ++i) {
        cin >> nums[i];
    }

    int mini = nums[0];
    for(int i=1;i<n;i++)
    {
        if(nums[i] < mini)
        mini = nums[i];
    }

    cout << mini << endl;
}

File: DSA/akashyadav_590014126/Day4Question1.py
def find_smallest_element(nums):
    if not nums:
        return None  # checks if the list is empty or not
    smallest_val = nums[0]
    for num in nums[1:]:
        if num < smallest_val:
            smallest_val = num
    return smallest_val

# Example usage:
nums = [40, 10, 30, 77, 100, 99]
print(find_smallest_element(nums))  # Output: 10